packages/outbox-execute/scripts/exec.js

```diff
@@ -19,90 +19,59 @@ const l1Provider = new providers.JsonRpcProvider(process.env.L1RPC)
-   */
+  */
 
   if (!txnHash)
     throw new Error(
       'Provide a transaction hash of an L2 transaction that sends an L2 to L1 message'
     )
   if (!txnHash.startsWith('0x') || txnHash.trim().length != 66)
     throw new Error(`Hmm, ${txnHash} doesn't look like a txn hash...`)
-  /**
-   * Use wallets to create an arb-ts bridge instance
-   * We'll use bridge for its convenience methods around outbox-execution
-   */
-  const bridge = await Bridge.init(l1Wallet, l2Wallet)
-
+  
   /**
    * First, let's find the Arbitrum txn from the txn hash provided
    */
-  const initiatingTxnReceipt = await bridge.l2Provider.getTransactionReceipt(
-    txnHash
-  )
-
-  if (!initiatingTxnReceipt)
-    throw new Error(
-      `No Arbitrum transaction found with provided txn hash: ${txnHash}`
-    )
-
-  /**
-   * In order to trigger the outbox message, we'll first need the outgoing messages batch number and index; together these two things uniquely identify an outgoing message.
-   * To get this data, we'll use getWithdrawalsInL2Transaction, which retrieves this data from the L2 events logs
-   */
-
-  const outGoingMessagesFromTxn = await bridge.getWithdrawalsInL2Transaction(
-    initiatingTxnReceipt
-  )
-
-  if (outGoingMessagesFromTxn.length === 0)
-    throw new Error(`Txn ${txnHash} did not initiate an outgoing messages`)
+  const receipt = await l2Provider.getTransactionReceipt(txnHash)
+  const l2Receipt = new L2TransactionReceipt(receipt)
 
   /**
    * Note that in principle, a single transaction could trigger any number of outgoing messages; the common case will be there's only one.
    * For the sake of this script, we assume there's only one / just grad the first one.
    */
-  const { batchNumber, indexInBatch } = outGoingMessagesFromTxn[0]
+  const l2Network = await getL2Network(l2Provider)
+  const messages = await l2Receipt.getL2ToL1Messages(l1Wallet, l2Network)
+  const l2ToL1Msg = messages[0]
+
 
   /**
-   * We've got batchNumber and IndexInBatch in hand; but before we try to execute out message, we need to make sure it's confirmed! (It can only be confirmed after the dispute period; Arbitrum is an optimistic rollup after-all)
-   * Here we'll do a period check; once getOutGoingMessageState tells us our txn is confirm, we'll move on to execution
+   * before we try to execute out message, we need to make sure it's confirmed! (It can only be confirmed after the dispute period; Arbitrum is an optimistic rollup after-all)
+   * Here we'll do a period check; once L2ToL1MessageStatus tells us our txn is confirm, we'll move on to execution
    */
-  const outgoingMessageState = await bridge.getOutGoingMessageState(
-    batchNumber,
-    indexInBatch
-  )
-  console.log(
-    `Waiting for message to be confirmed: Batchnumber: ${batchNumber}, IndexInBatch ${indexInBatch}`
-  )
-  console.log(`Outgoing message state: ${OutgoingMessageState[outgoingMessageState]}`)
 
+  const outgoingMessageState = await l2Receipt.status
```

Comment:
What we want here is `l2ToL1Msg.status()` (l2ToL1Msg is an L2ToL1MessageReader), not `l2Receipt.status` (l2Receipt is just an ordininary ethers-js tx receipt). Also we'll want to update outgoingMessageState each iteration in the while loop

Reply:
Updated